environment.cpp

```diff
@@ -49,6 +51,15 @@ bool operator== (const environment& lhs, const environment& rhs)
   return ref_t_values && current_t_value && cue_distrib && env_function;
 }
 
+void environment::switch_name_current_function()
+{
+    if(get_name_current_function() == 'A')
+        m_name_current_function = 'B';
+    else if(get_name_current_function() == 'B')
+        m_name_current_function = 'A';
+    else throw;
```

Comment:
Know you can add messages to thrown exceptions. Also, i will need to check, but I think you need to `catch` a thrown exception for it to show up and also specify that if the exception is catched you `abort()` the program. I will take care of this.